cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftService.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.NamespaceSummary;
+import io.cdap.cdap.api.metrics.Metrics;
+import io.cdap.cdap.etl.common.Constants;
+import io.cdap.cdap.etl.proto.v2.ETLConfig;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Class to interact with the DraftStore
+ */
+public class DraftService {
+  private static final Logger LOG = LoggerFactory.getLogger(DraftService.class);
+  private final Metrics metrics;
+  private final DraftStore store;
+
+  public DraftService(TransactionRunner context, Metrics metrics) {
+    this.store = new DraftStore(context);
+    this.metrics = metrics;
+    if (metrics == null) {
+      LOG.error("Metrics collector was not injected into StudioService");
```

Comment:
Log as warn. If you log as error, it should halt the flow.

Reply:
done